src/huggingface_hub/repository.py

```diff
@@ -441,6 +442,7 @@ def __init__(
         revision: Optional[str] = None,
         private: bool = False,
         skip_lfs_files: bool = False,
+        hf_api: Optional[HfApi] = None,
```

Comment:
I agree with @LysandreJik that `client` would be a much better name here for the argument.